cli/cmd/suppressions.go

```diff
@@ -63,4 +68,181 @@ func init() {
+	if len(discardedSuppressions) >= 1 {
+		cli.OutputHuman(color.YellowString("#### Discarded legacy suppressions\n"))
+		for _, suppression := range discardedSuppressions {
+			err := cli.OutputJSON(suppression)
+			if err != nil {
+				return
+			}
+		}
+	}
+}
+
+func convertSupCondition(supConditions []string, fieldKey string,
+	policyIdExceptionsTemplate []string) api.PolicyExceptionConstraint {
+	if len(supConditions) >= 1 && slices.Contains(
+		policyIdExceptionsTemplate, fieldKey) {
+
+		var condition []any
+		// verify for aws:
+		// if "ALL_ACCOUNTS" OR "ALL_REGIONS" is in the suppression condition slice
+		// verify for gcp:
+		// if "ALL_ORGANIZATIONS" OR "ALL_PROJECTS" is in the suppression condition slice
+		// if so we should ignore the supplied conditions and replace with a wildcard *
+		if (slices.Contains(supConditions, "ALL_ACCOUNTS") && fieldKey == "accountIds") ||
+			(slices.Contains(supConditions, "ALL_REGIONS") && fieldKey == "regionNames") {
+			condition = append(condition, "*")
+		} else if (slices.Contains(supConditions, "ALL_ORGANIZATIONS") && fieldKey == "organizations") ||
+			(slices.Contains(supConditions, "ALL_PROJECTS") && fieldKey == "projects") {
+			condition = append(condition, "*")
+		} else if fieldKey == "resourceNames" {
+			condition = convertResourceNamesSupConditions(supConditions)
+		} else if fieldKey == "resourceName" {
+			// resourceName singular is specific to GCP
+			condition = convertGcpResourceNameSupConditions(supConditions)
+		} else {
+			condition = convertToAnySlice(supConditions)
+		}
+
+		return api.PolicyExceptionConstraint{
+			FieldKey:    fieldKey,
+			FieldValues: condition,
+		}
+	}
+	return api.PolicyExceptionConstraint{}
+}
+
+func convertResourceNamesSupConditions(supConditions []string) []any {
+	var conditions []any
+	for _, condition := range supConditions {
+		ok := arn.IsARN(condition)
+		if ok {
+			parsedEntry, _ := arn.Parse(condition)
+			condition = parsedEntry.Resource
+		}
+		conditions = append(conditions, condition)
+	}
+	return conditions
+}
+
+func convertGcpResourceNameSupConditions(supConditions []string) []any {
+	var conditions []any
+	for _, condition := range supConditions {
+		// skip this logic if we already have a wildcard
+		if condition != "*" {
+			// It appears that for GCP, the resourceName field for policy exceptions is in fact expecting
+			// users to provider the full GCP resource_id.
+			// Example resourceId: //compute.googleapis.com/projects/gke-project-01-c8403ba1/zones/us-central1-a/instances/squid-proxy
+			// This was not the case for legacy suppressions and in most cases it's unlikely that the
+			// users will have provided this. Instead, we are more likely to have
+			// the resource name provided. To cover this scenario we prepend the resource name
+			// from the legacy suppression with "*/" to make it match the resource name while
+			// wildcarding the rest of the resourceId
+			condition = "*/" + condition
```

Comment:
were you able to double check this syntax a dev env.?

Reply:
Good point. I need to circle back an test this as I wasn't able to yesterday on dev5 or dev7

Reply:
Confirmed on dev8 that the suppression format is working as expected. The only concern would be if a user has already provided the full resource ID. However I don't believe this format was used anywhere in the legacy compliance checks.